src/app/src/main/kotlin/org/imperial/mrc/hint/controllers/ProjectsController.kt

```diff
@@ -31,12 +31,25 @@ class ProjectsController(private val session: Session,
         return SuccessResponse(project).asResponseEntity()
     }
 
+    @PostMapping("/user/{email}/project/")
+    @ResponseBody
+    fun cloneProjectToUser(@RequestParam("parentProjectId") parentProjectId: Int,
+                           @PathVariable("email") email: String): ResponseEntity<String> {
+
+        val userId = userLogic.getUser(email)?.id ?: throw UserException("userDoesNotExist")
+        val currentProject = projectRepository.getProject(parentProjectId)
```

Comment:
We could pass the currrent user id to the repository here, to check that they are permitted to get or clone the source project.